Added functionality to merge the chapters into a single archive

JLommen opened the pull request:
I've added a flag that users can set using -m to merge all the chapters into an individual cbz file instead of each chapter being its own file. I thought this functionality would be useful for people like me who read on an e-reader where a large number of files becomes difficult to manage. I tested the new functionality as well as the original functionality and it worked as expected on my local system. Thanks for your work on the original!

JLommen committed:
Added functionality to merge the chapters into a single archive

avchikar97 commented on main.py:
```diff
@@ -44,14 +48,25 @@ def cbz_folder(folder_name:str):
+           files=os.listdir(os.path.join(os.getcwd(),folder_name))
+           for fname in files:
+            file_extension = Path(os.path.join(os.getcwd(),folder_name,fname)).suffix
+            shutil.copy(os.path.join(os.path.join(os.getcwd(),folder_name,fname)),os.path.join(os.getcwd(),"working_merge/",(str(NUMBER_FILES_MERGED)+file_extension)))
```
I think something is wrong with the destination path here. I see `tachiyomi_path/working_merge/` was created on line 126, so is the second argument of `shutil.copy(...)` supposed to be `os.path.join(tachiyomi_path,"working_merge",(str(NUMBER_FILES_MERGED)+file_extension))`?



Without the change I got the following error in my little toy folder 
```
Traceback (most recent call last):
  File "main.py", line 128, in <module>
    cbz_search(tachiyomi_path, directory_list)
  File "main.py", line 107, in cbz_search
    cbz_search(folder_name, directory_only_list)
  File "main.py", line 107, in cbz_search
    cbz_search(folder_name, directory_only_list)
  File "main.py", line 107, in cbz_search
    cbz_search(folder_name, directory_only_list)
  File "main.py", line 111, in cbz_search
    cbz_folder(folder_name)
  File "main.py", line 65, in cbz_folder
    shutil.copy(os.path.join(os.path.join(os.getcwd(),folder_name,fname)),os.path.join(os.getcwd(),"working_merge/",(str(NUMBER_FILES_MERGED)+file_extension)))  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\shutil.py", line 418, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\test\\tachiyomi\\downloads\\MangaDex\\Berserk\\working_merge/0.jpg'
```

avchikar97 commented:
I'm using a little toy tachiyomi folder to test this represented below:
```
D:.
│   main.py
│
└───tachiyomi
    ├───downloads
    │   └───MangaDex
    │       ├───Berserk
    │       │   ├───Chapter 1
    │       │   │       hori (1).jpg
    │       │   │       hori (2).jpg
    │       │   │
    │       │   └───Chapter 2
    │       │           hori (3).jpg
    │       │           hori (4).jpg
    │       │
    │       └───Hibike
    │           ├───Chapter 3
    │           │       hori (7).jpg
    │           │       hori (8).jpg
    │           │
    │           └───Chapter 4
    │                   hori (5).jpg
    │                   hori (6).jpg
    │
    └───local
        └───Yamada to Kase
            ├───Chapter 1
            │       hori (10).jpg
            │       hori (9).jpg
            │
            └───Chapter 3
                    hori (11).jpg
                    hori (12).jpg
```

Just to make sure I get the reason for this, is the purpose of the `--merge` flag to make `merged.cbz` which contains all the .jpg files in the tachiyomi folder (merging all the folders)?

Also you're welcome and I'm happy that somebody else is getting some use out of this little hacked-together script!

JLommen committed:
Corrected error with path in nested directories when in merge mode. Additional edits made to support Windows and Linux system paths

JLommen commented:
Ah yes you're totally right! Sorry for not catching that before submitting, I was not testing with nested directories and I see the problem. I noticed that if I just replaced the code at line 67 with tachiyomi_path it would cause issues if the program was launched with a relative path for the -p argument instead of the absolute path. I went ahead and added the absolute path as a global variable and set its value immediately on entry to the main function so it could be available to new functions should more functionality be added later. I initially just added it to line 65 (now 67) but noticed that it would still have errors when running on Windows instead of Linux. I've now tested it on Linux and Windows with both nested and non-nested directories and it appears to function as expected. 
And yes to clarify the reason I wanted this functionality is because I often receive manga volumes as a single directory with subdirectories for each chapter and wanted a way to quickly convert the entire volume with each chapter into a single cbz.
Please let me know if any other strange behavior is noted, I'll be sure to correct it asap.

avchikar97 reviewed (approved):
Lgtm
